File: Visual Studio Product Team/Official Visual Studio 2008 VB.NET Samples/46871-Official Visual Studio 2008 VB.NET Samples/LINQ to TerraServer Provider Sample/VB/LinqToTerraServer/InvalidQueryException.vb
Public Class InvalidQueryException
    Inherits Exception

    Private _message As String

    Public Sub New(ByVal message As String)
        Me._message = message & " "
    End Sub

    Public Overrides ReadOnly Property Message() As String
        Get
            Return "The client query is invalid: " & _message
        End Get
    End Property
End Class

File: Visual Studio Product Team/Official Visual Studio 2008 C# Samples/46335-Official Visual Studio 2008 C# Samples/LINQ to TerraServer Provider Sample/C#/LinqToTerraServerProvider/InvalidQueryException.cs
using System;

namespace LinqToTerraServerProvider
{
    [Serializable]
    public class InvalidQueryException : System.Exception
    {
        private string message;

        public InvalidQueryException() { }

        public InvalidQueryException(string message)
        {
            this.message = message + " ";
        }

        public override string Message
        {
            get
            {
                return "The client query is invalid: " + message;
            }
        }
    }
}
